x-pack/plugins/features/server/index.ts

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { PluginInitializerContext } from '../../../../src/core/server';
+import { Plugin } from './plugin';
+
+// These exports are part of public Features plugin contract, any change in signature of exported
+// functions or removal of exports should be considered as a breaking change. Ideally we should
+// reduce number of such exports to zero and provide everything we want to expose via Setup/Start
+// run-time contracts.
+export { uiCapabilitiesRegex } from './feature_schema';
```

Comment:
For what it's worth, we should probably move `uiCapabilitiesRegex` into `src/core`, as UI Capabilities are a part of OSS now. You can leave it here for the time being, but if we're concerned about API surface, that's one way to reduce it 😄.

**question**: your comment about reducing these exports to zero doesn't include the types we export below, right? We can still export types from our plugins outside of the lifecycle methods?

Reply:
> For what it's worth, we should probably move uiCapabilitiesRegex into src/core, as UI Capabilities are a part of OSS now. You can leave it here for the time being, but if we're concerned about API surface, that's one way to reduce it 

Yeah, I tried to do that several times while working on this, but failed to do so every time :laughing: I just can't find a proper place for it, it was kind of confusing that it sits in the core, but is not used there at all and it's not clear how exactly it's used while reading core code. If you can help me to find a justification I'll happily move that to the core :slightly_smiling_face:

Reply:
> question: your comment about reducing these exports to zero doesn't include the types we export below, right? We can still export types from our plugins outside of the lifecycle methods?

Correct!

Reply:
> Yeah, I tried to do that several times while working on this, but failed to do so every time 😆 I just can't find a proper place for it, it was kind of confusing that it sits in the core, but is not used there at all and it's not clear how exactly it's used while reading core code. If you can help me to find a justification I'll happily move that to the core 🙂


It doesn't exist today, but I could see this being useful when we [build the list of capabilities](https://github.com/elastic/kibana/blob/d9c34caf956eefa5504ce6c98ee96ade718380f8/src/legacy/server/capabilities/capabilities_mixin.ts#L43). Currently, plugins can provide any capabilities they'd like, even if they violate the regex. We're just trusting that they don't right now.


It's potentially also useful when we [resolve capabilities](https://github.com/elastic/kibana/blob/d9c34caf956eefa5504ce6c98ee96ade718380f8/src/legacy/server/capabilities/resolve_capabilities.ts#L32). Core might want to prevent registered Capabilities Modifiers from returning a capability which violates the regex. In theory, the modifiers shouldn't be adding or removing any capabilities; rather, they should just be flipping the boolean flags...

Reply:
Ah, good to know, thanks. Yeah, it makes a lot of sense to have it in core then.

Reply:
Moved to the core.

Reply:
Nah, had to revert and move back to the `Features` plugin. There is something weird happening when we require regex from `ui/capabilities` in server code. It's probably expected, but I'll let platform team to handle that when they want to move that to core :)

Reply:
Thanks for trying!